src/QSymbolicsBase/literal_objects.jl

```diff
@@ -92,4 +92,4 @@ symbollabel(x::SZero) =  "𝟎"
 basis(x::SZero) = nothing
 
 Base.show(io::IO, x::SZero) = print(io, symbollabel(x))
-Base.iszero(x::SymQObj) = isa(x, SZero)
+Base.iszero(x::Union{SymQObj, Symbolic{Number}, Symbolic{Complex}}) = isa(x, SZero)
```

Comment:
I think this is piracy. Why is this needed?

Reply:
My bad about the piracy. It was intended for when we process `*` for `SScaled` objects, although it's kinda of a hackish way to deal with this issue. Basically, I was getting errors before when I would multiply an inner product (which is a subtype of `Symbolic{Complex}`) or a real number plus an inner product (subtype of `Symbolic{Number}`), as `iszero` was not defined for `SBraKet`. I'll fix this.